network-config-analyzer/ProtocolNameResolver.py

```diff
@@ -0,0 +1,94 @@
+                                     14: 'EMCON', 15: 'XNET', 16: 'CHAOS', 17: 'UDP', 18: 'MUX', 19: 'DCNMEAS',
+                                     20: 'HMP', 21: 'PRM', 22: 'XNSIDP', 23: 'TRUNK1', 24: 'TRUNK2', 25: 'LEAF1',
+                                     26: 'LEAF2', 27: 'RDP', 28: 'IRTP', 29: 'ISOTP4', 30: 'NETBLT', 31: 'MFENSP',
+                                     32: 'MERITINP', 33: 'DCCP', 34: '3PC', 35: 'IDPR', 36: 'XTP', 37: 'DDP',
+                                     38: 'IDPRCMTP', 39: 'TP', 40: 'IL', 41: 'IPv6', 42: 'SDRP', 43: 'IPv6Route',
+                                     44: 'IPv6Frag', 45: 'IDRP', 46: 'RSVP', 47: 'GRE', 48: 'DSR', 49: 'BNA', 50: 'ESP',
+                                     51: 'AH', 52: 'INLSP', 53: 'SWIPEdeprecated', 54: 'NARP', 55: 'MOBILE', 56: 'TLSP',
+                                     57: 'SKIP', 58: 'ICMPv6', 59: 'IPv6NoNxt', 60: 'IPv6Opts', 62: 'CFTP',
+                                     64: 'SATEXPAK', 65: 'KRYPTOLAN', 66: 'RVD', 67: 'IPPC', 69: 'SATMON', 70: 'VISA',
+                                     71: 'IPCV', 72: 'CPNX', 73: 'CPHB', 74: 'WSN', 75: 'PVP', 76: 'BRSATMON',
+                                     77: 'SUNND', 78: 'WBMON', 79: 'WBEXPAK', 80: 'ISOIP', 81: 'VMTP', 82: 'SECUREVMTP',
+                                     83: 'VINES', 84: 'IPTM', 85: 'NSFNETIGP', 86: 'DGP', 87: 'TCF', 88: 'EIGRP',
+                                     89: 'OSPFIGP', 90: 'SpriteRPC', 91: 'LARP', 92: 'MTP', 93: 'AX25', 94: 'IPIP',
+                                     95: 'MICPdeprecated', 96: 'SCCSP', 97: 'ETHERIP', 98: 'ENCAP', 100: 'GMTP',
+                                     101: 'IFMP', 102: 'PNNI', 103: 'PIM', 104: 'ARIS', 105: 'SCPS', 106: 'QNX',
+                                     107: 'AN', 108: 'IPComp', 109: 'SNP', 110: 'CompaqPeer', 111: 'IPXinIP',
+                                     112: 'VRRP', 113: 'PGM', 115: 'L2TP', 116: 'DDX', 117: 'IATP', 118: 'STP',
+                                     119: 'SRP', 120: 'UTI', 121: 'SMP', 122: 'SMdeprecated', 123: 'PTP',
+                                     124: 'ISISoverIPv4', 125: 'FIRE', 126: 'CRTP', 127: 'CRUDP', 128: 'SSCOPMCE',
+                                     129: 'IPLT', 130: 'SPS', 131: 'PIPE', 132: 'SCTP', 133: 'FC', 134: 'RSVPE2EIGNORE',
+                                     135: 'MobilityHeader', 136: 'UDPLite', 137: 'MPLSinIP', 138: 'manet', 139: 'HIP',
+                                     140: 'Shim6', 141: 'WESP', 142: 'ROHC', 143: 'Ethernet'}
+
+    @staticmethod
+    def get_protocol_name(protocol_number: int) -> str:
+        """
+        :param protocol_number: Protocol number
+        :return: The protocol name
+        :rtype: str
+        """
+        if protocol_number < 1 or protocol_number > 255:
+            raise Exception('Protocol number must be in the range 1-255')
+
+        return ProtocolNameResolver._protocol_number_to_name_dict[protocol_number]
```

Comment:
```suggestion
        return ProtocolNameResolver._protocol_number_to_name_dict.get(protocol_number, str(protocol_number))
```
dict's [] operator throws an exception if the key to search is not found in the dict.
`dict.get()` is safer, and can be used to return the string representation of the integer like we did before.

Reply:
Done